### PR TITLE
Rename beb to eventmesh in ec

### PR DESCRIPTION
**Pull request by friedrichwilken**
Rename `beb` and `ems` to `eventmesh` at multiple (but not all, yet) places in the code. Also, satisfy the linter by shortening line length, reducing function complexity and so on.

### CODE REVIEW

File: installation/resources/crds/eventing/subscriptions.eventing.kyma-project.io.crd.yaml [hunk 1]
```diff
@@ -3,6 +3,7 @@ kind: CustomResourceDefinition
 metadata:
   annotations:
     controller-gen.kubebuilder.io/version: v0.11.1
+  creationTimestamp: null
```
Comment (grischperl) [inline installation/resources/crds/eventing/subscriptions.eventing.kyma-project.io.crd.yaml:6]: Is this needed?

File: components/eventing-controller/api/v1alpha1/fixtures_test.go [hunk 2]
```diff
@@ -70,7 +71,7 @@ func newDefaultSubscription(opts ...eventingtesting.SubscriptionV1alpha1Opt) *v1
 		o(newSub)
 	}
 
-	// remove nats specific field in beb case
+	// remove nats specific field in eventmesh case
 	if newSub.Status.EmsSubscriptionStatus != nil {
```
[the remark this answers is not in the document]
Comment (friedrichwilken) [reply components/eventing-controller/api/v1alpha1/fixtures_test.go:75]: yes, one by one

File: components/eventing-controller/config/crd/external/apirules-gateway-kyma-project-io.yaml [hunk 3]
```diff
@@ -47,75 +47,80 @@ spec:
...
-                description: Rules represents collection of Rule to apply
+                description: Represents the array of Oathkeeper access rules to be applied.
                 items:
-                  description: Rule .
                   properties:
                     accessStrategies:
-                      description: Set of access strategies for a single path
+                      description: Specifies the list of access strategies.
+                        All strategies listed in [Oathkeeper documentation](https://www.ory.sh/docs/oathkeeper/pipeline/authn) 
+                        are supported.
```
[the remark this answers is not in the document]
Comment (friedrichwilken) [reply components/eventing-controller/config/crd/external/apirules-gateway-kyma-project-io.yaml:63]: that's the magic of `make copy-external-crds`

File: installation/resources/crds/eventing/subscriptions.eventing.kyma-project.io.crd.yaml (hunk 1 above)
Comment (friedrichwilken) [reply installation/resources/crds/eventing/subscriptions.eventing.kyma-project.io.crd.yaml:6]: that's the magic of `make copy-external-crds`

File: components/eventing-controller/testing/test_helpers.go [hunk 4]
```diff
@@ -33,6 +35,8 @@ const (
 	EventSourceUnclean       = "s>o>*u*r>c.e"
 	EventSourceClean         = "source"
 
+	EventMeshProtocl = "BEB"
```
[the remark this answers is not in the document]
Comment (friedrichwilken) [reply components/eventing-controller/testing/test_helpers.go]: fixed

**Review by grischperl — APPROVED** [review]

**Pull request merged**